src/core/LfoController.cpp

```diff
@@ -102,15 +103,35 @@ void LfoController::updateValueBuffer()
 	ValueBuffer *amountBuffer = m_amountModel.valueBuffer();
 	int amountInc = amountBuffer ? 1 : 0;
 	float *amountPtr = amountBuffer ? &(amountBuffer->values()[ 0 ] ) : &amount;
+	Oscillator::WaveShape waveshape = 
+			static_cast<Oscillator::WaveShape>(m_waveModel.value());
 
 	for( float& f : m_valueBuffer )
 	{
-		const float currentSample = m_sampleFunction != nullptr
-			? m_sampleFunction( phase )
-			: m_userDefSampleBuffer->userWaveSample( phase );
+		float currentSample = 0;
+		switch (waveshape)
+		{
+			case Oscillator::WaveShape::WhiteNoise:
+				if (absFraction(phase) < absFraction(phasePrev))
+				{
+					// Resample when phase period has completed
+					m_heldSample = m_sampleFunction( phase );
```

Comment:
I interpret the code convention to be:
`m_sampleFunction(phase);`

...for all instances of `( phase )`